packages/dds/merge-tree/src/client.ts

```diff
@@ -472,22 +486,44 @@ export class Client extends TypedEventEmitter<IClientEvents> {
 
 	private applyObliterateRangeOp(opArgs: IMergeTreeDeltaOpArgs): void {
 		assert(
-			opArgs.op.type === MergeTreeDeltaType.OBLITERATE,
+			opArgs.op.type === MergeTreeDeltaType.OBLITERATE ||
+				opArgs.op.type === MergeTreeDeltaType.OBLITERATE_SIDED,
 			0x866 /* Unexpected op type on range obliterate! */,
 		);
 		const op = opArgs.op;
 		const clientArgs = this.getClientSequenceArgs(opArgs);
-		const range = this.getValidOpRange(op, clientArgs);
-
-		this._mergeTree.obliterateRange(
-			range.start,
-			range.end,
-			clientArgs.referenceSequenceNumber,
-			clientArgs.clientId,
-			clientArgs.sequenceNumber,
-			false,
-			opArgs,
+		assert(
+			op.pos1 !== undefined && op.pos2 !== undefined,
+			"Obliterate op missing start or end",
 		);
+		if (this._mergeTree.options?.mergeTreeEnableSidedObliterate) {
+			/*
+			const _start: InteriorSequencePlace =
+				typeof op.pos1 === "object"
+					? { pos: op.pos1.pos ?? 0, side: op.pos1.before ? Side.Before : Side.After }
+					: { pos: op.pos1, side: Side.Before };
+			const _end: InteriorSequencePlace =
+				typeof op.pos2 === "object"
+					? { pos: op.pos2.pos ?? 0, side: op.pos2.before ? Side.Before : Side.After }
```

Comment:
Not necessarily something that has to be dealt with here, but should the end position default to 0 or the length of the string? I think if it defaults to 0, there will be issues since zero-length obliterate doesn't work yet.

Reply:
pos1 and pos2 can no longer be undefined, so we actually don't need defaults here. Removed.